### PR TITLE
add bypass func

### PR DESCRIPTION
**Pull request by natebrennand**

### CODE REVIEW

File: transforms/transforms.go [hunk 1]
```diff
@@ -262,3 +262,49 @@ func GroupBy(identifier RowIdentifier) optimus.TransformFunc {
 		return nil
 	}
 }
+
+// RowFilter is meant to return `true` if a section is meant to be filtered out.
+type RowFilter func(optimus.Row) bool
+
+// passthroughTable is a super simple optimus table that just passes rows through
+type passthroughTable struct {
```
Comment (peggyl) [inline transforms/transforms.go]: Why is this necessary, or why doesn't just passing a `chan optimus.Row` work?

File: transforms/transforms.go [hunk 2]
```diff
@@ -262,3 +262,49 @@ func GroupBy(identifier RowIdentifier) optimus.TransformFunc {
...
+	return p.c
+}
+func (p passthroughTable) Err() error {
+	return nil
+}
+func (p passthroughTable) Stop() {}
+
+// BypassTransforms effectively wraps a slice of transform funcs with a gate to conditionally
+// apply the transforms only if they match the filter.
+func BypassTransforms(doBypass RowFilter, optionalTransforms []optimus.TransformFunc) optimus.TransformFunc {
```
Comment (peggyl) [inline transforms/transforms.go]: I still find the API a little unintuitive: given a set of objects and a function I want to apply to some of them, I would expect most APIs to logically be `doApply` rather than `doNotApply` (seems more common).

File: transforms/transforms.go [hunk 3]
```diff
@@ -262,3 +262,49 @@ func GroupBy(identifier RowIdentifier) optimus.TransformFunc {
...
+	outboundRows := make(chan optimus.Row)
+
+	// setup an optimus pipeline to process rows through all the provided optional transforms
+	passthroughChan := make(chan optimus.Row)
```
Comment (peggyl) [inline transforms/transforms.go]: Similar comment to `passthroughTable` - I'm not too familiar with optimus architecture, but why is the tempTable necessary as opposed to applying the transform on each row in the returned func?

Comment (natebrennand) [inline transforms/transforms.go]: It is messy to apply a `TransformFunc` due to their obtuse API: `func(in <-chan Row, out chan<- Row) error`. It is not meant to return until the input channel has been closed. [This is messy to do correctly](https://github.com/Clever/optimus/blob/v3.7.0/optimus.go#L70-L136).

Creating the table that has everything passed through seemed like a cleaner solution to avoid reimplementing the [`start`](https://github.com/Clever/optimus/blob/9caeeff3bb4965f14d6392fbed2a751a0d81c225/optimus.go#L70-L136) function.

File: transforms/transforms.go (hunk 2 above)
Comment (natebrennand) [inline transforms/transforms.go]: I could've sworn you gave me the opposite feedback during sprint demos... https://github.com/Clever/optimus/pull/56/commits/4f840f63e8e29925ae52ee439ad1a7e696137542

Comment (peggyl) [inline transforms/transforms.go]: That was a preference for the func name matching the first argument for consistency. I think the other way around is a more intuitive API, but that could be personal experience/preference, so not going to block on it.

File: transforms/transforms.go (hunk 3 above)
Comment (peggyl) [inline transforms/transforms.go]: 👍

**Comment by peggyl** [comment]
Seems fine as long as it's working in the pipeline workers.
